### PR TITLE
Internal: standardize format for a11y integration test naming

### PR DESCRIPTION
**Pull request by dangerismycat**
I noticed recently that our accessibility integration test file naming is chaos…different casing, whitespace, and names that have no bearing on the route they hit. This makes it difficult to see which routes have coverage, as demonstrated by the seven duplicate test files I deleted, as well as a test for a route we no longer support (without a redirect, this is basically testing a blank page). 

The new structure follows the route path, converting `/` to `__` but otherwise staying true to the URL (snake_case, not camelCase or PascalCase). Though this adds a bit more pain whenever we adjust paths (something we shouldn't be doing frequently anyway), this makes it very easy to find the relevant test file for a path, and to know what name a new test file should use.

This PR also reworks (simplifies) our validation script to take advantage of the consistent file format.

I tried something a bit fancier where the tests would use their own file name to determine the path in `page.goto`, but I don't think that's possible in Playwright tests.

It would also be really cool if we could obviate the test files altogether with a script that basically creates test files on the fly from everything in `docs/pages`, but that's an improvement for another day.

I also discovered a couple of lil other fixes, which I'll call out in comments.

### CODE REVIEW

File: docs/docs-components/siteIndex.js [hunk 1]
```diff
@@ -185,7 +185,7 @@ const siteIndex: $ReadOnlyArray<siteIndexType> = [
           'Grammar',
           'Inclusive language',
           'UI elements',
-          'Syntax structure',
+          'Syntax and structure',
```
Comment (dangerismycat) [inline docs/docs-components/siteIndex.js:188]: This appeared to be a typo

File: docs/markdown/foundations/content_standards/syntax_and_structure.md [hunk 2]
```diff
@@ -1,27 +1,27 @@
 ---
-title: Syntax and Structure
+title: Syntax and structure
```
Comment (dangerismycat) [inline docs/markdown/foundations/content_standards/syntax_and_structure.md:2]: This was the only change I made in this file. The other changes were prettier

File: docs/redirects.js [hunk 3]
```diff
@@ -138,6 +138,11 @@ const misc = [
     destination: '/get_started/developers/contributing/development_process',
     permanent: true,
   },
+  {
```
Comment (dangerismycat) [inline docs/redirects.js:141]: This is the only new redirect. The others are just alphabetizing

File: playwright/accessibility/foundations__messaging__overview.spec.mjs [hunk 4]
```diff
@@ -2,7 +2,7 @@
 import { test } from '@playwright/test';
 import expectAccessiblePage from './expectAccessiblePage.mjs';
 
-test('Data viz palette check', async ({ page }) => {
-  await page.goto('/foundations/data_visualization/palette');
+test('Messaging overview accessibility check', async ({ page }) => {
```
Comment (dangerismycat) [inline playwright/accessibility/foundations__messaging__overview.spec.mjs:5]: git's being weird here. This wasn't converting a test to a different path — probably from a duplicate test I deleted

File: scripts/validateAccessibilitySpecs.js [hunk 5]
```diff
@@ -31,50 +31,35 @@ const getAllSitePaths = (index) => {
...
+  const a11ySpecFiles = (await globby([specFileGlob])).map((file) =>
+    file
+      .toLowerCase()
+      // remove the playwright/accessibility/ prefix
+      .replace(/^playwright\/accessibility\//, '')
+      // remove the .spec.mjs suffix
+      .replace(/\.spec\.mjs$/, ''),
   );
 
+  const pagesWithoutA11ySpecFiles = routes.filter((page) => !a11ySpecFiles.includes(page));
```
Comment (dangerismycat) [inline scripts/validateAccessibilitySpecs.js:53]: The consistent format makes this check much simpler — convert the route path from the index to match the file name format and do a simple comparison

File: playwright/accessibility/foundations__forms__structure_and_behavior.spec.mjs [hunk 6]
```diff
@@ -3,6 +3,6 @@ import { test } from '@playwright/test';
 import expectAccessiblePage from './expectAccessiblePage.mjs';
 
 test('Structure and behavior Accessibility check', async ({ page }) => {
-  await page.goto('/foundations/forms/composition_+_behavior');
```
Comment (dangerismycat) [inline playwright/accessibility/foundations__forms__structure_and_behavior.spec.mjs:6]: this was just wrong 😬

**Review by AlbertCarreras — DISMISSED** [review]

**Comment by dangerismycat** [comment]
@rlingineni The thing to keep in mind is that these tests are not testing components, they're testing routes. We actually need to add accessibility tests for our components themselves as the current tests can't "see" any of the Sandpack examples, as they're rendered in iframes. So basing the test name on the route rather than the component makes the most sense to me.

**Pull request merged**